### ISSUE TITLE
[feature] CUVID hardware decoding option

### ISSUE THREAD

**Issue by BuyMyMojo** [paulpacifico/shutter-encoder#115]
When doing cpu encodes I have found I can slightly improve encode speeds by letting my GPU do the decoding first, on Nvidia you can add `-c:v h264_cuvid` or `-c:v hevc_cuvid` before the input file, so `ffmpeg -c:v h264_cuvid -i file.mp4 ...`

having cuvid as an option in the hardware acceleration drop-down would be nice to speed up encoding that little bit while sticking with CPU quality.

**Comment by BuyMyMojo** [comment]
Oh and when doing nvenc hardware encodes on a compatible hardware decoded format doing `-hwaccel cuda -hwaccel_output_format cuda -i input` might be more efficient since it processes everything in vram without moving frame data from cpu to gpu

**Comment by paulpacifico** [comment]
You have the cuda option from top left cog icon which open the 'Settings' window then select it from the 'Set GPU decoding'.

**Comment by BuyMyMojo** [comment]
I didn't see that option, after taking a look I don't think it is actually working, when manually running the ffmpeg command generated from this error appears when using both libx264 and h264_nvenc and there is 0% video decode load in task manager.

![WindowsTerminal_sQQrvlm5fl](https://user-images.githubusercontent.com/46948241/216816550-db318cd3-c33c-4634-8079-2e624700182a.png)

Adding `-c:v h264_cuvid` to the command fixes this and starts using the GPU for decode.

Remmoving `-hwaccel cuda` but keeping cuvid doesn'y appear to affect anything.

having these 3 options however `-hwaccel cuda -hwaccel_output_format cuda -c:v h264_cuvid` and removing `-pix_fmt` completely removes the small 10% cpu load left over when doing nvenc encodes(on my 5600x at least, this will vary) but will not work when doing cpu encodes

To summarise;
- the cuda decode option isn't currently doing anything
- `-c:v X_cuvid` work with all encoding options
  - The possible options for cuvid are: `vp8_cuvid vp9_cuvid vc1_cuvid mpeg4_cuvid mpeg2_cuvid mpeg1_cuvid mjpeg_cuvid hevc_cuvid h264_cuvid av1_cuvid` which will depend on the card for indavidual codec support
- for optimal GPU encodes you need all 3 `-hwaccel cuda -hwaccel_output_format cuda -c:v X_cuvid` to elimate cpu usage
  - when using the above 3 options `-pix_fmt` breaks it (-hwaccel_output_format cuda outputs in a cuda colour space which saves as NV12)
